xwiki-platform-core/xwiki-platform-uiextension/xwiki-platform-uiextension-api/src/main/java/org/xwiki/uiextension/internal/WikiUIExtension.java

```diff
@@ -158,13 +162,16 @@ private Object before(boolean inline) throws RenderingException
             throw new RenderingException("Failed to get ui extension document", e);
         }
         Map<String, Object> uixContext = new HashMap<>();
-        uixContext.put(WikiUIExtension.CONTEXT_UIX_DOC_KEY, document.newDocument(xcontext));
-        uixContext.put(WikiUIExtension.CONTEXT_UIX_INLINE_KEY, inline);
+        uixContext.put(CONTEXT_UIX_DOC_KEY, document.newDocument(xcontext));
+        uixContext.put(CONTEXT_UIX_INLINE_KEY, inline);
 
         // Remember the previous uix context to restore it
-        Map<String, Object> previousUIXContext = (Map<String, Object>) xcontext.get(WikiUIExtension.CONTEXT_UIX_KEY);
-        // Put the UIX context in the XWiki context
-        xcontext.put(WikiUIExtension.CONTEXT_UIX_KEY, uixContext);
+        Map<String, Object> previousUIXContext = (Map<String, Object>) xcontext.get(CONTEXT_UIX_KEY);
+        // Put the UIX context in the XWiki context. Note that this is deprecated and using the UIX templates is
+        // preferred.
+        xcontext.put(CONTEXT_UIX_KEY, uixContext);
+        // Put the UIX context in the velocity context "uix" key.
+        this.velocityManager.getVelocityContext().put(CONTEXT_UIX_KEY, uixContext);
```

Comment:
This is somehow a breaking change if `uix` was already set in the velocity context.

Reply:
Variables should be set in the ScriptContext and never in the VelocityContext.

Reply:
VelocityContext replaced by ScriptContext. In addition I've fixed the context restoration. Now each context is respectively reset with its own previous values (before my last commit, both context where reset with the same previous value).